### PR TITLE
[BACKLOG-21186] Fixed sorting step metrics

### PR DESCRIPTION
**Pull request by ricksjames**
[BACKLOG-21186] Fixed sorting step metrics

### CODE REVIEW

**Comment by ricksjames** [comment]
@pentaho/ackbar please review

File: core/src/main/java/org/pentaho/di/core/row/value/ValueMetaBase.java [hunk 1]
```diff
@@ -3628,8 +3628,8 @@ public int compare( Object data1, Object data2 ) throws KettleValueException {
         // if (isStorageBinaryString() && identicalFormat &&
         // storageMetadata.isSingleByteEncoding()) return
         // compareBinaryStrings((byte[])data1, (byte[])data2); TODO
-        String one = getString( data1 );
-        String two = getString( data2 );
+        String one = getString( data1 ).trim();
+        String two = getString( data2 ).trim();
```
Comment (mkambol) [inline core/src/main/java/org/pentaho/di/core/row/value/ValueMetaBase.java]: This'll change more than just sorting of step metrics.  Do we know for sure that leading and tailing white space should be ignored in all cases?

**Comment by ricksjames** [comment]
@pentaho/ackbar please review again :)

File: ui/src/main/java/org/pentaho/di/ui/core/widget/TableView.java [hunk 2]
```diff
@@ -153,13 +153,9 @@
   private Listener lsFocusInTabItem;
 
   private int sortfield;
-  private int sortfieldLast;
```
Comment (kurtwalker) [inline ui/src/main/java/org/pentaho/di/ui/core/widget/TableView.java]: I'm confused why these fields were removed??

**Comment by ricksjames** [comment]
@pentaho/ackbar please review

**Pull request merged**